astropy/visualization/wcsaxes/coordinate_helpers.py

```diff
@@ -186,11 +186,12 @@ def grid(self, draw_grid=True, grid_type=None, **kwargs):
 
         self.grid_lines_kwargs.update(kwargs)
 
-        if self.grid_lines_kwargs["visible"]:
-            if not draw_grid:
-                self.grid_lines_kwargs["visible"] = False
+        if isinstance(draw_grid, bool):
+            self.grid_lines_kwargs["visible"] = draw_grid
+        elif draw_grid is None:
+            draw_grid = True
```

Comment:
I missed this during my previous review, but this branch is effectively doing nothing now since `draw_grid` is not used later. It might be why CI is failing again ? I'm sorry if I introduced a mistake with my refactor proposal ! Please let me know if you need a hand fixing this again.

Reply:
Thanks for pointing that out! My bad for missing it earlier. I'll fix it right away.

Reply:
note that we can still avoid rebinding `draw_grid` here

Reply:
Right, so the branch becomes:
`if isinstance(draw_grid, bool):
            self.grid_lines_kwargs["visible"] = draw_grid
        elif draw_grid is None:
            self.grid_lines_kwargs["visible"] = True
        else:
            raise TypeError(f"draw_grid expected a boolean or None, got {draw_grid}")`
I missed this completely, thanks again for pointing it out